.travis.yml

```diff
@@ -21,24 +21,66 @@ linux32_addons:
         - linux-libc-dev:i386
         - libc6-dev-i386
 
+linux64_cpp17addons:
+    addons: &linux64cpp17
+      apt:
+        sources:
+        - ubuntu-toolchain-r-test
+
+
+linux64_cpp17addons_py36:
+    addons: &linux64cpp17py36
+      apt:
+        sources:
+        - ubuntu-toolchain-r-test
+        - gcc-7
+        - g++-7
+
 # Set empty values for allow_failures to work
-env:
+env: TEST_COMMAND=$TRAVIS_BUILD_DIR/ci/build_and_run_tests.sh
+python:
 
 matrix:
   fast_finish: true
   include:
     - os: linux
-      env: EMSCRIPTEN=ON
+      env: EMSCRIPTEN=ON TEST_COMMAND=$TRAVIS_BUILD_DIR/ci/build_and_run_tests.sh
       addons: *linux64
     - os: linux
       compiler: clang
       addons: *linux64
     - os: linux
       compiler: gcc
-      env: ARCH=x86 CMAKE_EXTRA=-DHAVE_LIBM=/lib32/libm.so.6
+      env: ARCH=x86 CMAKE_EXTRA=-DHAVE_LIBM=/lib32/libm.so.6 TEST_COMMAND=$TRAVIS_BUILD_DIR/ci/build_and_run_tests.sh
```

Comment:
The TEST_COMMAND environment variable is a way to run two different tests.  I could not find a better way to run two different test suites.  If you have any examples of repos that do this in a nicer way, I would be happy to learn a more efficient way and refactor this.

in this case TEST_COMMAND is not redundant because the `env:` block is overwriting the global value.